presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestTablePartitioningInsertInto.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tests.hive;
+
+import com.google.inject.Inject;
+import com.teradata.tempto.Requirement;
+import com.teradata.tempto.RequirementsProvider;
+import com.teradata.tempto.configuration.Configuration;
+import com.teradata.tempto.fulfillment.table.MutableTableRequirement;
+import com.teradata.tempto.fulfillment.table.MutableTablesState;
+import com.teradata.tempto.fulfillment.table.hive.HiveTableDefinition;
+import com.teradata.tempto.query.QueryResult;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.tests.TestGroups.HIVE_CONNECTOR;
+import static com.facebook.presto.tests.TestGroups.SMOKE;
+import static com.teradata.tempto.Requirements.compose;
+import static com.teradata.tempto.fulfillment.table.hive.InlineDataSource.createResourceDataSource;
+import static com.teradata.tempto.query.QueryExecutor.query;
+import static com.teradata.tempto.query.QueryType.UPDATE;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestTablePartitioningInsertInto
+        extends HivePartitioningTest
+        implements RequirementsProvider
+{
+    @Inject
+    private MutableTablesState mutableTablesState;
```

Comment:
We tend to avoid injectable private fields. We think of injection frameworks as a layer that helps assemble your objects, but it should be possible to do so manually if one wants to (e.g., in a unit test). If something is accessible to the injection framework, it should be accessible to regular code.